source-git-monorepos/README.md

```diff
@@ -50,21 +89,367 @@ packages:
+```
+
+Example:
+
+```shell
+$ packit source-git update-dist-git --dist-git-store rpms/ src/copr
+```
+
+### `packit source-git update-source-git`
+
+Let's not change the interface of this command and don't support the usecase
+to update a single source-git monorepo from multiple dist-git repositories in
+order to avoid complications. Expect that multiple commands are going to be
+executed if updates need to be taken from multiple dist-git repositories.
+
+On the other hand, the command needs to be updated to support a source-git
+repo as a target when the source-git repo is a monorepo. In this case only the
+paths of the sub-package need to be considered during the update. `DIST_GIT`
+should match one of the `downstream_package_name`s in `.distro/source-git.yaml`.
+
+### `packit source-git status`
+
+The command should check the sync status of a sub-package from a source-git
+monorepo and the corresponding dist-git repo. `downstream_package_name` should
+match `basename DIST_DIS`.
+
+### Hardly
+
+When opening an MR in a source-git monorepo, Hardly should figure out which
+dist-git packages are going to be impacted by the changed (by looking at
+`package_paths`), and open mirror-MRs in the dist-git repositories of those
+packages.
+
+CI results from all mirror-MRs should be synced back to the source-git MR.
+
+**Open question**: Is the behavior above good enough? Can there be source-git
+changes happening outside the package paths, but which influence the dist-git
+content of the package?
+
+## Impact on Packit
+
+All things that follow assume that [the `metadata` key was dropped].
+
+Additionally to what was presented above, a new property of job-objects in
+`.packit.yaml` needs to be introduced, to be used to specify which packages
+the job should handle when triggered. The property would be called `packages`
+and would be a list of package-object names. If a job-object has no such key,
+all packages of a monorepo should be handled.
+
+Example:
+
+```yaml
+upstream_project_url: https://pagure.io/copr/copr.git
+packages:
+  copr-backend:
+    specfile_path: backend/copr-backend.spec
+    package_paths:
+      - backend
+  copr-frontend:
+    specfile_path: frontend/copr-frontend.spec
+    package_paths:
+      - frontend
+  copr-cli:
+    specfile_path: cli/copr-cli.spec
+    package_paths:
+      - cli
+jobs:
+  - job: copr_build
+    trigger: pull_request
+    targets:
+      - fedora-all
+      - epel-8
+    packages:
+      - copr-frontend
+      - copr-backend
+  - job: copr_build
+    trigger: pull_request
+    targets:
+      - fedora-all
+    packages:
+      - copr-cli
+```
+
+### packit init
+
+No changes are required for this command. `.packit.yaml` is expected to be
+crafted manually if the upstream repository is a monorepo.
+
+### packit srpm
+
+The command should receive a new optional argument, `[PACKAGE...]`, to define
+zero or more packages for which an SRPM should be built.
```

Comment:
Ditto here, having a possibility to call this also from within sub-dir would be plus.

Reply:
Although plain `packit` commands prefer to guess their arguments from their context, context might be ambiguous so I try to follow the "explicit is better than implicit" mantra to avoid confusion.

If the current proposal is going to be followed than calling `packit srpm` from a sub-dir would result in building an SRPM for all the packages in the monorepo (Packit would look up the config in the repo-root, and b/c there is not `PACKAGE` argument provided, it will build an SRPM for all the packages defined in the config.

Of course, we could say, that if the current sub-dir is a `package_path` for a package in the config, then build just that SRPM. Which is fine, if there is only one `package_path` in a package, but becomes more complicated if there are two or more. And to avoid this complication and possible user confusion, I would prefer to stay explicit :)

Reply:
The only example I know is in LLVM. For their daily builds, the do something source-git-like, and for `clang` they [fetch 2 paths](https://github.com/kwk/llvm-project/blob/0eacd884dd720b7a2156fb84df910388c5677b33/clang/clang.spec#L88) from their monorepo.